src/features/import/utils/createPreviewData.ts

```diff
@@ -0,0 +1,56 @@
+import { ZetkinPersonImportOp } from './prepareImportOperations';
+import { ColumnKind, Sheet } from './types';
+
+export default function createPreviewData(
```

Comment:
What is the purpose of this function? Doesn't it just duplicate what `prepareImportOperations` do, but for a single row?

I know we talked about this, and how code might be reused by breaking out the row-by-row logic from `prepareImportOperations` into a separate function, and then using that function inside of `prepareImportOperations`.

Am I right in interpreting this code as the first half of that, i.e. duplicating the logic, but then this function is not used inside of `prepareImportOperations`, effectively causing the same code to be in two places?

Reply:
As far as I know, Ziggi added more logic in prepareImportOperations that doesn't need in my function. So I made separate function for it and created its tests. Also thought that it is not actually prepare importing operation but just creating data for preview. Maybe I totally misunderstood things here😵

Reply:
It's completely possible to do what you've done here. I'm just asking because it goes against what we discussed when we talked about this. Let me explain in writing how I remember that conversation.

I recommended that you use `prepareImportOperations` because it already does what you need here, and it's well tested and consistent with what will eventually be sent to the server (because it actually decides what gets sent). That's all you could wish for in a preview, IMO.

By creating a separate function, you lose those qualities. You make up for the loss of test coverage by writing new tests, but I see no reason to not just use `prepareImportOperations` directly then.

The one problem I raised when we discussed this was that it wouldn't be optimized for performance, since `prepareImportOperations` looks at all rows, not just the one we're trying to preview. But like I said then, I don't think that will actually be a problem, and if it is, the way to solve it is by refactoring `prepareImportOperations` as described above (by breaking out the row-by-row logic to a separate function).

Reply:
Yeah I used `prepareImportOperations` for preview data and it won't be problematic for performance as you said before. When we discussed about this, I remember that it would be nice to have something like `prepareImportOperationsForRow` that returns only one data for preview.
 But it didn't have priority as you mentioned however it is something nice to have, so we decided to go with `prepareImportOperations` first and then work on `prepareImportOperationsForRow`(createPreviewData in this case) if I have time. I made a note like that but maybe I got it wrong. 

After I finished preview, I found time that I could work on this `prepareImportOperationsForRow` so that is why I created this function. Sorry if I got it wrong

Reply:
I don't think it was a mistake to create a separate function, but I would have _used that function_ from within `prepareImportOperations`, like so (pseudo-code):

```ts
export function prepareImportOperationsForRow(sheet, rowIndex) {
  // Do the logic for a row
}

export default function prepareImportOperations(sheet) {
  // This is a simplification of the logic, which would likely need to be more complicated
  return sheet.rows.map((_, rowIndex) => prepareImportOperationsForRow(sheet, rowIndex));
}
```

This way, the new function would already be tested by the tests for `prepareImportOperations`, because if the new function doesn't work, neither would `prepareImportOperations`.

I can explain this more thoroughly tomorrow.